---
title: "TestingTCGA"
output: pdf_document
date: "2026-01-21"
---

```{r, include=FALSE, message=FALSE, warning=FALSE}
# BiocManager::install('TCGAbiolinks')
library('TCGAbiolinks')
library('tidyverse')
```

Use GDCquery to get meta data.
```{r}
query <- GDCquery(
  project = "TCGA-BRCA", 
  data.category = "Transcriptome Profiling",
  data.type = "Gene Expression Quantification",
  workflow.type = "STAR - Counts"
)
```

This gives us a list containing all the metadata GDC matching the arguments in GDCquery()

 ├─ [[1]]  # Typically the first "category" (data.category or project level)
 │   ├─ [[1]]  # First data table of files (like the metadata table of files available)
 │   ├─ [[2]]  # Maybe additional info (depends on TCGAbiolinks version)
 ├─ [[2]]  # If multiple categories/projects, second element

So to view the metadata collected:

```{r}
meta_data <- query[[1]][[1]]
head(meta_data)

colnames(meta_data)
```
Some notes about the "cases" column:

TCGA-AB-1234-01A-11R-1234-07

First 12 characters is patient ID: TCGA-AB-1234

01A: sample type (01: primary tumor; 11: Solid Tissue Normal)

The rest: vial, portion, analyte, plate, center info...

If we are getting paired data (paired normal and tumor cell):


```{r}
# First create a new column containing only patient ID
meta_data <- meta_data %>% 
  mutate(patient_id = substr(cases, 1, 12))

# Second, create another dataset containing only patients with paired data
paired_data <- meta_data %>% 
  group_by((patient_id)) %>%  # This is so we could check by patient
  filter(all(c("Primary Tumor", "Solid Tissue Normal") %in% sample_type)) %>% 
        # For each patient, is data available for both tumor and normal samples?
  ungroup() # Remove the grouping => have separate rows for normal and tumor again

# Some patients might have multiple tumor samples, and multiple normal samples:
paired_data %>% 
  filter(sample_type == "Primary Tumor") %>%
  group_by(patient_id) %>% 
  summarise(n_tumor = n()) %>%
  print()

paired_data %>% 
  filter(sample_type == "Solid Tissue Normal") %>%
  group_by(patient_id) %>% 
  summarise(n_normal = n()) %>%
  print()
```


NOT SURE WHY YET: need to separate into 2 tables: normal and tumor
```{r}
normal_data <- paired_data %>% 
  filter(sample_type == "Solid Tissue Normal")

tumor_data <- paired_data %>% 
  filter(sample_type == "Primary Tumor")
```

NOT SURE WHY YET: create a joined table containing file ID for download
```{r}
file_table <- normal_data %>% 
  select(patient_id, normal_file_id = file_id) %>% 
  left_join(
    tumor_data %>% 
      select(patient_id, tumor_file_id = file_id), 
    by = "patient_id"
  )

```



